File: packages/room-server/src/database/entities/datasheet.operation.entity.ts
/**
 * APITable <https://github.com/apitable/apitable>
 * Copyright (C) 2022 APITable Ltd. <https://apitable.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

import { Column, Entity } from 'typeorm';
import { BaseEntity } from 'shared/entities/base.entity';

/**
 * Workbench-Datasheet Operation
 */
@Entity('datasheet_operation')
export class DatasheetOperationEntity extends BaseEntity {

  @Column({
    name: 'op_id',
    nullable: true,
    unique: true,
    comment: 'operation ID',
    length: 50,
  })
  opId!: string;

  @Column({
    name: 'dst_id',
    nullable: true,
    comment: 'datasheet ID(related#datasheet#dst_id)',
    length: 50,
  })
  dstId?: string;

  @Column({
    name: 'action_name',
    nullable: true,
    comment: 'action name',
    length: 255,
  })
  actionName?: string;

  @Column('json', { name: 'actions', nullable: true, comment: 'action collection' })
  actions!: object;

  @Column({
    name: 'type',
    nullable: true,
    comment: 'action type(1:JOT,2:COT)',
    unsigned: true,
  })
  type?: number;

  @Column({
    name: 'member_id',
    nullable: true,
    comment: 'operating member ID(related#organization_member#id)',
  })
  memberId?: string;

  @Column({
    name: 'revision',
    comment: 'revision',
    unsigned: true,
    default: () => 0,
  })
  revision!: number;
}

File: backend-server/application/src/main/java/com/apitable/workspace/entity/DatasheetOperationEntity.java
/*
 * APITable <https://github.com/apitable/apitable>
 * Copyright (C) 2022 APITable Ltd. <https://apitable.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package com.apitable.workspace.entity;

import java.io.Serializable;
import java.time.LocalDateTime;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

/**
 * <p>
 * Workbench - datasheet operation table
 * </p>
 *
 * @author Mybatis Generator Tool
 */
@Data
@Builder(toBuilder = true)
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
@EqualsAndHashCode
@TableName(keepGlobalPrefix = true, value = "datasheet_operation")
public class DatasheetOperationEntity implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * Primary key
     */
    @TableId(value = "id", type = IdType.ASSIGN_ID)
    private Long id;

    /**
     * Operation ID
     */
    private String opId;

    /**
     * Datasheet ID(link#xxxx_datasheet#dst_id)
     */
    private String dstId;

    /**
     * Operation name
     */
    private String actionName;

    /**
     * Collection of operations
     */
    private String actions;

    /**
     * Type(1:JOT,2:COT)
     */
    private Integer type;

    /**
     * Action member ID(link#xxxx_organization_member#id)
     */
    private Long memberId;

    /**
     * Version No
     */
    private Long revision;

    /**
     * Delete tag(0:No,1:Yes)
     */
    @TableLogic
    private Boolean isDeleted;

    /**
     * Create User
     */
    @TableField(fill = FieldFill.INSERT)
    private Long createdBy;

    /**
     * Last Update User
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Long updatedBy;

    /**
     * Create time
     */
    private LocalDateTime createTime;

    /**
     * Update time
     */
    private LocalDateTime updateTime;

    /**
     * Create Time
     */
    private LocalDateTime createdAt;

    /**
     * Update Time
     */
    private LocalDateTime updatedAt;

}
